gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -8,21 +8,58 @@ import com.android.build.gradle.TestExtension
 import com.android.build.gradle.TestedExtension
 import com.android.build.gradle.api.BaseVariant
 import org.gradle.api.GradleException
-import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.plugins.AppliedPlugin
 import org.gradle.api.plugins.PluginManager
 import org.gradle.api.provider.Provider
 import org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask
 import org.jetbrains.kotlin.gradle.plugin.KaptExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
 import org.jetbrains.kotlin.gradle.plugin.KotlinPluginWrapper
 import org.jetbrains.kotlin.gradle.plugin.PLUGIN_CLASSPATH_CONFIGURATION_NAME
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.io.File
 import java.util.Locale.US
 import java.util.concurrent.atomic.AtomicBoolean
 
-open class AnvilPlugin : Plugin<Project> {
-  override fun apply(project: Project) {
+open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
+
+  override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean {
+    return kotlinCompilation.target.project.plugins.hasPlugin(AnvilPlugin::class.java)
```

Comment:
I was also surprised that the plugin is applied to all our projects. How does that work ? Why do we have to check this? Isn't the compiler plugin only applied to projects using the plugin ?

Reply:
It's a sanity check. You can still apply the Gradle plugin to a module and the compiler plugin to single Kotlin compilation tasks. I our case we apply it to all.